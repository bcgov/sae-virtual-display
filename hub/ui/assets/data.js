window.vdiApps = [
  {
    name: 'notebook',
    logo: 'notebook',
    label: 'Jupyterlab',
    container: 'quay.io/ikethecoder/vdi-session-notebook:develop',
  },
  {
    name: 'browser',
    logo: 'browser',
    label: 'Chrome Browser',
    container: 'quay.io/ikethecoder/vdi-session-browser:develop',
  },
  {
    name: 'rstudio',
    logo: 'rstudio',
    label: 'RStudio (1.2.5019)',
    container:
      'quay.io/ikethecoder/vdi-session-rstudio:d852f663b1af2fbdfb41bd7fd1a319b0a74d2772',
  },
  {
    name: 'rstudio-edge',
    logo: 'rstudio',
    label: 'RStudio (EDGE)',
    container:
      'quay.io/ikethecoder/vdi-session-rstudio:a27fdcf55273f18e950352b7c8e973a0f7e1f45d',
  },
  {
    name: 'rstudio-dev',
    logo: 'rstudio',
    label: 'RStudio (DEV)',
    container:
      'quay.io/ikethecoder/vdi-session-rstudio:07de87a1965c1d9b3ce7a2a98ad3bd2cf04b6d7d',
  },
  {
    name: 'conda-navigator',
    logo: 'anaconda',
    label: 'Conda Navigator (Admin)',
    container:
      'quay.io/ikethecoder/vdi-session-navigator:a27fdcf55273f18e950352b7c8e973a0f7e1f45d',
  },
  {
    name: 'data-curator',
    logo: 'datacurator',
    label: 'Data Curator (Metadata)',
    container:
      'quay.io/ikethecoder/vdi-session-datacurator:1c231d3b6a58e6b2c9bd3468a623d8a5c3f35b1d',
  },
];
