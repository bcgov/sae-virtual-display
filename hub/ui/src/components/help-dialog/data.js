const data = [
  {
    id: 1,
    page: {
      body: '<p>This is <i>some</i> <strong>body</strong> content</p>',
    },
  },
];

export default data;
