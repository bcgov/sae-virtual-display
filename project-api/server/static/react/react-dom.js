Cannot find "/dist/react-dom.js" in react@0.14.3