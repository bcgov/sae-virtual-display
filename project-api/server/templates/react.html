<!DOCTYPE html>
<html>
    <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <link rel="stylesheet" type="text/css" href="/static/bootstrap/bootstrap.min.css"/>
        <script src="/static/bootstrap/jquery-3.4.1.min.js"></script>
        <script src="/static/bootstrap/bootstrap.min.js"></script>
        <title>DIP Self-Service</title>
    </head>
    <body>
            <div class="container">
                    <h1>Groups</h1>
                    <p>{{ groups }}</p>
                    <div id="root"></div>
                    <script src="/static/react/react.min.js"></script>
                    <script src="/static/react/react-dom.min.js"></script>
                    <script src="/static/react/browser.min.js"></script>
            
                    <script type="text/babel" src="/static/app/greeting.jsx"></script>
            </div>
    </body>
</html>
