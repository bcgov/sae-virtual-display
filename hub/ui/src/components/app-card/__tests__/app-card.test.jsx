import * as React from 'react';
import { cleanup, fireEvent, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';

import data from './data';
import AppCard from '../';
import Loading from '../loading';

const cachedLocation = window.location;

describe('app-card', () => {
  afterEach(cleanup);

  describe('app-card/index', () => {
    it('should render', () => {
      const { container } = render(<AppCard data={data} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render labels correctly', () => {
      const { getByText, rerender } = render(<AppCard data={data} />);
      expect(getByText('Idle')).toBeTruthy();

      rerender(<AppCard ready data={data} />);
      expect(getByText('Running')).toBeTruthy();

      rerender(<AppCard error data={data} />);
      expect(getByText('Failed')).toBeTruthy();
    });

    it('should launch a ready app', () => {
      const onLaunch = jest.fn();
      const { getByText } = render(
        <AppCard ready data={data} onLaunch={onLaunch} />,
      );
      fireEvent.click(getByText('Launch'));

      expect(onLaunch).toHaveBeenCalled();
    });

    it('should stop event bubbling on info button', () => {
      const onStart = jest.fn();
      const { getByText } = render(
        <AppCard data={{ ...data, container: '/test' }} onStart={onStart} />,
      );
      fireEvent.click(getByText('View Info'));
      expect(onStart).not.toHaveBeenCalled();
    });

    describe('app-card/in-progress', () => {
      it('should render loading state', () => {
        const { getByText } = render(
          <AppCard loading data={{ ...data, ready: false }} />,
        );
        expect(getByText('Loading...')).toBeTruthy();
      });
      it('should render progress bar', async () => {
        const onStartApp = jest.fn();
        const onClick = jest.fn();
        const { getByText } = render(
          <AppCard
            data={{ ...data, ready: false }}
            onClick={onClick}
            onStartApp={onStartApp}
          />,
        );
        fireEvent.click(getByText('Build Application'));

        expect(onStartApp).toHaveBeenCalled();
        expect(onClick).not.toHaveBeenCalled();
      });

      it('should render progress view when passed progress value', () => {
        const progressText = 'In Progress';
        const { getByText } = render(
          <AppCard
            data={{ ...data, ready: false }}
            progress={10}
            message={progressText}
          />,
        );
        expect(getByText(progressText)).toBeTruthy();
        expect(getByText('10% complete')).toBeTruthy();
      });
    });
  });

  describe('app-card/loading', () => {
    it('should render', () => {
      const { container, getAllByTestId } = render(<Loading />);

      expect(container.firstChild).toMatchSnapshot();
      expect(getAllByTestId('app-card-loading').length).toEqual(1);
    });

    it('should render total amount of rows requested', () => {
      const { getAllByTestId } = render(<Loading total={5} />);
      expect(getAllByTestId('app-card-loading').length).toEqual(5);
    });
  });

  describe('app-card/running', () => {
    const onClick = jest.fn();
    const onShutdown = jest.fn();
    const { getByText } = render(
      <AppCard ready data={data} onClick={onClick} onShutdown={onShutdown} />,
    );
    fireEvent.click(getByText('Shutdown'));

    expect(onShutdown).toHaveBeenCalled();
    expect(onClick).not.toHaveBeenCalled();
  });
});
