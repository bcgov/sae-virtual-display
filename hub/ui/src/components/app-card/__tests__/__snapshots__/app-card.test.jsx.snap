// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app-card app-card/index should render 1`] = `
.c4 {
  color: #36B37E;
  display: inline-block;
  fill: #FFFFFF;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  line-height: 1;
}

.c4 > svg {
  max-height: 100%;
  max-width: 100%;
  overflow: hidden;
  pointer-events: none;
  vertical-align: bottom;
}

.c4 stop {
  stop-color: currentColor;
}

.c10 {
  height: 16px;
  width: 16px;
  color: currentColor;
  display: inline-block;
  fill: #FFFFFF;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  line-height: 1;
}

.c10 > svg {
  height: 16px;
  width: 16px;
  max-height: 100%;
  max-width: 100%;
  overflow: hidden;
  pointer-events: none;
  vertical-align: bottom;
}

.c10 stop {
  stop-color: currentColor;
}

.c6 {
  max-width: 100%;
  height: auto;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: absolute;
  top: 5px;
  right: 5px;
  border-radius: 3px;
  -webkit-transition: opacity 0.25s ease-out;
  transition: opacity 0.25s ease-out;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 1rem;
  position: relative;
  margin-bottom: 1rem;
  background-color: #fff;
  border-radius: 3px;
  -webkit-transition: all 0.2s ease-in;
  transition: all 0.2s ease-in;
  box-shadow: 0 1px 1px rgba(9,30,66,0.25),0 0 1px 0 rgba(9,30,66,0.31);
}

.c1 .c2 {
  opacity: 0;
}

.c1 {
  cursor: pointer;
}

.c1:hover {
  box-shadow: 0 8px 16px -4px rgba(9,30,66,0.25),0 0 1px rgba(9,30,66,0.31);
}

.c1:hover .c2 {
  opacity: 1;
}

.c7 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin: 0.25rem 0;
}

.c7 p {
  margin-bottom: 0.5rem;
}

.c5 {
  margin-right: 1rem;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c9 a {
  padding-left: 0 !important;
}

.c9 p {
  margin: 0 10px 0 0;
}

.c9 p + a {
  padding-left: 8px;
}

.c8 {
  font-size: 18px;
  font-weight: 600;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #36B37E;
}

.c8 > span {
  margin-left: 10px;
}

.c0:hover .c8 {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div
  class="c0 c1"
  data-testid="app-card"
>
  <div
    class="c2 c3"
  >
    <button
      class="css-shc4i4"
      type="button"
    >
      <span
        class="css-j8fq0c"
      >
        <span
          class="css-t5emrf"
        >
          Launch
        </span>
        <span
          class="css-8xpfx5"
        >
          <span
            class="c4"
          >
            <svg
              focusable="false"
              height="24"
              role="presentation"
              viewBox="0 0 24 24"
              width="24"
            >
              <path
                d="M14 11.414l-3 3V18h3v2H6.006A1.999 1.999 0 0 1 4 17.994V10h2v3h3.586l3-3H12c-.552 0-1-.444-1-1 0-.552.444-1 1-1h3a.997.997 0 0 1 1 1v3c0 .552-.444 1-1 1-.552 0-1-.444-1-1v-.586zM16 18v2h2c1.105 0 2-.888 2-2v-2h-2v1.497c0 .27-.225.503-.503.503H16zM8 6V4H6c-1.105 0-2 .888-2 2v2h2V6.503c0-.27.225-.503.503-.503H8zm10 2h2V6c0-1.112-.895-2-2-2h-2v2h1.497c.278 0 .503.233.503.503V8zm-8-4h4v2h-4V4zm8 6h2v4h-2v-4z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </span>
      </span>
    </button>
  </div>
  <div
    class="c5"
  >
    <img
      alt="R Studio Logo"
      class="c6"
      src="/images/rstudio.png"
      title="R Studio"
      width="80"
    />
  </div>
  <div
    class="c7"
  >
    <div
      class=""
    >
      <h2
        class="c8"
      >
        R Studio
        <span
          class="css-vlkpln"
        >
          <span
            class="css-q5kkvr"
          >
            Idle
          </span>
        </span>
      </h2>
      <small>
        Application has not been started yet
        Running about 1 month
      </small>
      <div
        class="c9"
      >
        <p>
          R Studio is an integrated development environment for r, a programming language for statistical computing and graphics.
        </p>
        <a
          class="css-1hzvll6"
          href="https://quay.io/containername:test"
          type="button"
        >
          <span
            class="css-j8fq0c"
          >
            <span
              class="css-t5emrf"
            >
              View Info
            </span>
            <span
              class="css-8xpfx5"
            >
              <span
                class="c10"
              >
                <svg
                  focusable="false"
                  height="24"
                  role="presentation"
                  viewBox="0 0 24 24"
                  width="24"
                >
                  <g
                    fill="currentColor"
                  >
                    <path
                      d="M19.005 19c-.003 0-.005.002-.005.002l.005-.002zM5 19.006c0-.004-.002-.006-.005-.006H5v.006zM5 4.994V5v-.006zM19 19v-6h2v6.002A1.996 1.996 0 0 1 19.005 21H4.995A1.996 1.996 0 0 1 3 19.006V4.994C3 3.893 3.896 3 4.997 3H11v2H5v14h14zM5 4.994V5v-.006zm0 14.012c0-.004-.002-.006-.005-.006H5v.006zM11 5H5v14h14v-6h2v6.002A1.996 1.996 0 0 1 19.005 21H4.995A1.996 1.996 0 0 1 3 19.006V4.994C3 3.893 3.896 3 4.997 3H11v2zm8 0v3a1 1 0 0 0 2 0V4a1 1 0 0 0-1-1h-4a1 1 0 0 0 0 2h3z"
                    />
                    <path
                      d="M12.707 12.707l8-8a1 1 0 1 0-1.414-1.414l-8 8a1 1 0 0 0 1.414 1.414z"
                    />
                  </g>
                </svg>
              </span>
            </span>
          </span>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`app-card app-card/loading should render 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 1rem;
  position: relative;
  margin-bottom: 1rem;
  background-color: #fff;
  border-radius: 3px;
  -webkit-transition: all 0.2s ease-in;
  transition: all 0.2s ease-in;
  height: 84px;
  overflow: hidden;
}

.c0 .sc-fzXfLT {
  opacity: 0;
}

.c0 {
  cursor: pointer;
}

.c0:hover {
  box-shadow: 0 8px 16px -4px rgba(9,30,66,0.25),0 0 1px rgba(9,30,66,0.31);
}

.c0:hover .sc-fzXfLT {
  opacity: 1;
}

.c0:hover {
  cursor: default;
  box-shadow: none;
  background: #fff;
}

.c0 > div:first-child {
  width: 100px;
  height: 100%;
  margin-right: 20px;
  border-radius: 3px;
  background: #EBECF0;
}

.c0 > div:last-child {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0 div:last-child > div {
  width: 75%;
  height: 15px;
  margin: 5px 0;
  display: block;
  border-radius: 3px;
  background: #EBECF0;
}

.c0 div:last-child > div:first-child {
  width: 24%;
  height: 20px;
}

.c0 div:last-child > div:nth-child(2) {
  width: 45%;
}

<div
  class="sc-fzXfLU c0"
  data-testid="app-card-loading"
>
  <div />
  <div>
    <div />
    <div />
    <div />
  </div>
</div>
`;
